R/plotSeries.R

```diff
@@ -0,0 +1,330 @@
+                }
+            }
+        }
+    }
+
+    # If feature_data exists, add feature_data in addition to series_data
+    if(exists("feature_data")){
+        returned_list <- list(sample_data = sample_data, feature_data = feature_data)
+    }# If it does not exist, just add the series_data
+    else{
+        returned_list <- list(sample_data = sample_data)
+    }
+    return(returned_list)
+}
+
+.melt_series_data <- function(assay, sample_data, feature_data){
+    
+    # Melt assay table 
+    melted_data <- as.data.frame(assay) %>% pivot_longer(colnames(assay), names_to = "sample", values_to = "Y")
+    
+    # Add series data to the data frame
+    melted_data <- cbind(melted_data, sample_data)
+    
+    # If feature_data is not null
+    if( !is.null(feature_data) ){
+        # Loops through feature_data
+        for( i in 1:ncol(feature_data) ){
+            # Assigns feature data to data points
+            melted_data <- cbind(melted_data, feature = rep(feature_data[[i]], nrow(melted_data)/length(feature_data[[i]])))
+            # Renames the column
+            names(melted_data)[names(melted_data) == "feature"] <- names(feature_data)[i]
+        }
+    }
+    
+    return(melted_data)
+}
+
+.series_plotter <- function(plot_data,
+                            xlab = NULL,
+                            ylab = NULL,
+                            colour_by = NULL,
+                            colour_by_title = NULL,
+                            linetype_by = NULL,
+                            linetype_by_title = NULL,
+                            add_legend = TRUE,
+                            point_alpha = 1,
+                            point_size = 2,
+                            line_alpha = 1,
+                            line_size = 1,
+                            ...){
+    
+    # Creates a "draft" of a plot
+    plot_out <- ggplot(plot_data, aes_string(x = "X", y = "Y")) +
+        labs(x = xlab, y = ylab)
+    
+    # Fetch arguments of line
+    line_args <- .get_line_args(colour_by = colour_by,
+                                linetype_by = linetype_by,
+                                size_by = NULL,
+                                alpha = line_alpha,
+                                size = line_size)
+    
+    line_args$args$mapping$group <- sym("colour_by")
+    
+    # Adds arguments to the plot
+    plot_out <- plot_out +
+        do.call(geom_line, line_args$args)
+    
+    # resolve the colours
+    plot_out <- .resolve_plot_colours(plot_out,
+                                      plot_data$colour_by,
+                                      colour_by,
+                                      fill = TRUE)
+    plot_out <- .resolve_plot_colours(plot_out,
+                                      plot_data$colour_by,
+                                      colour_by,
+                                      fill = FALSE)
```

Comment:
This only need to be done once, since the fill argument is not applicable for geom_line, if I remember correctly. So probably best to keep just the `fill = FALSE` call.

Reply:
Done only once, fill = FALSE left